pkg/sql/distsql_physical_planner.go

```diff
@@ -191,10 +189,10 @@ func NewDistSQLPlanner(
 		gossip:               gw,
 		sqlInstanceDialer:    sqlInstanceDialer,
 		nodeHealth: distSQLNodeHealth{
-			gossip:             gw,
-			connHealthSystem:   connHealthCheckerSystem,
-			connHealthInstance: instanceConnHealthChecker,
-			isAvailable:        isAvailable,
+			gossip:                gw,
+			connHealth:            sqlInstanceDialer.ConnHealthTryDial,
+			connHealthWithResolve: sqlInstanceDialer.ConnHealthTryDialWithResolve,
+			isAvailable:           isAvailable,
```

Comment:
Maybe we can eliminate this whole structure or just eliminate `isAvailable`. And probably some other cleanups present in #127293

Reply:
I need to pick up that PR again at some point (or someone could finish it). I can take a look again next week when I should have some time.

Reply:
If this PR looks correct, the next step is to use the work done in https://github.com/cockroachdb/cockroach/pull/128283 and see if we can use only `checkInstanceHealth` and remove `checkSystem`, thereby eliminating the usage of `kvNodeDialer`. If this all turns out well, we might only need a small subset of your PR  (I think just simplifying what `isAvailable` is doing here). I think I can see for your PR later since I'm in the context.